perf/ring.go

```diff
@@ -93,16 +93,22 @@ func createPerfEvent(cpu int, opts ReaderOptions) (int, error) {
 		watermark = 1
 	}
 
-	attr := unix.PerfEventAttr{
-		Type:        unix.PERF_TYPE_SOFTWARE,
-		Config:      unix.PERF_COUNT_SW_BPF_OUTPUT,
-		Bits:        unix.PerfBitWatermark,
-		Sample_type: unix.PERF_SAMPLE_RAW,
-		Wakeup:      uint32(watermark),
+	var attr *unix.PerfEventAttr
+
+	if opts.Attr != nil {
+		attr = opts.Attr
```

Comment:
This doesn't enforce that opts.Watermark, etc. are zero if opts.Attr is not nil. PerfEventAttr is really a superset of ReaderOptions, so it doesn't seem right to add it to the latter type. What if w add a new constructor `NewReaderFromAttr(*PerfEventAttr)` instead?

Reply:
right, sounds less confusing